vald/evm/evm.go

```diff
@@ -486,6 +458,41 @@ func (mgr Mgr) GetTxReceiptIfFinalized(chain nexus.ChainName, txID common.Hash,
 	return txReceipt, nil
 }
 
+func (mgr Mgr) GetMultipleTxReceiptsIfFinalized(chain nexus.ChainName, txIDs []common.Hash, confHeight uint64) ([]rs.Result[*geth.Receipt], error) {
+	client, ok := mgr.rpcs[strings.ToLower(chain.String())]
+	if !ok {
+		return nil, fmt.Errorf("rpc client not found for chain %s", chain.String())
+	}
+
+	keyvals := []interface{}{"chain", chain.String()}
+
+	results, err := client.TransactionReceipts(context.Background(), txIDs)
```

Comment:
sort results by block number decs so that the cache can be better utilized when determining fanality later?

Reply:
i thought about that before, it's a single LatestFinalizedBlockNumber regardless of sorting or not. wdyt?

Reply:
What do you mean?

Reply:
updated the isTxReceiptFinalized, set to cache after query